pkgs/development/python-modules/manifold3d/default.nix

```
@@ -0,0 +1,87 @@
+{
+  lib,
+  buildPythonPackage,
+  python,
+  fetchFromGitHub,
+  scikit-build-core,
+  cmake,
+  ninja,
+  nanobind,
+  pkg-config,
+  numpy,
+  clipper2,
+  tbb,
+  glm,
+  gtest,
+  pytestCheckHook,
+  trimesh,
+}:
+
+let
+  # archived library, but manifold3d has removed this on master
+  thrust-src = fetchFromGitHub {
+    owner = "NVIDIA";
+    repo = "thrust";
+    rev = "refs/tags/2.1.0";
+    hash = "sha256-U9WgRZva7R/bNOF5VZTvIwIQDQDD3/bRO08j2TPLl9Q=";
+    fetchSubmodules = true;
+  };
+
+in
+
+buildPythonPackage rec {
+  pname = "manifold3d";
+  version = "2.5.1";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "elalish";
+    repo = "manifold";
+    rev = "v${version}";
```

```suggestion
    rev = "refs/tags/v${version}";
```
This is preferred in order to prevent branch/tag mismatch.

---

verified with `--check`